test/freenet/clients/http/utils/UriFilterProxyHeaderParserTest.java

```diff
@@ -0,0 +1,342 @@
+    testUriPrefixMatchesExpected(
+        "8889",
+        "127.0.0.1",
+        "",
+        "",
+        MultiValueTable.from(new String[]{ "x-forwarded-host" }, new String[]{ "foo:8888" }),
+        "http://127.0.0.1:8889");
+  }
+
+  @Test
+  public void disallowedProxyWithStandardPortIsIgnoredIfPortIsNotStandard() {
+    testUriPrefixMatchesExpected(
+        "8889",
+        "127.0.0.1",
+        "",
+        "",
+        MultiValueTable.from(
+            new String[]{ "x-forwarded-host" },
+            new String[]{ "127.0.0.1:8888" }),
+        "http://127.0.0.1:8889");
+  }
+
+  // ignored uri values
+  // no such bindToHost
+  @Test
+  public void disallowedUriHostIsIgnored() {
+    testUriPrefixMatchesExpected(
+        "8888",
+        "127.0.0.1",
+        "",
+        "foo",
+        MultiValueTable.from(new String[]{ "x-forwarded-host" }, new String[]{ "foo" }),
+        "http://127.0.0.1:8888");
+  }
+
+  @Test
+  public void disallowedUriHostWithStandardPortIsIgnored() {
+    testUriPrefixMatchesExpected(
+        "8888",
+        "127.0.0.1",
+        "",
+        "foo:8888",
+        MultiValueTable.from(new String[]{ "x-forwarded-host" }, new String[]{ "foo:8888" }),
+        "http://127.0.0.1:8888");
+  }
+
+  // port mismatch
+  @Test
+  public void disallowedUriWithAllowedHostButDisallowedPortIsIgnored() {
+    testUriPrefixMatchesExpected(
+        "8888",
+        "127.0.0.1,foo",
+        "",
+        "foo:8889",
+        MultiValueTable.from(new String[]{ "x-forwarded-host" }, new String[]{ "foo:8889" }),
+        "http://127.0.0.1:8888");
+  }
+
+  private void testUriPrefixMatchesExpected(
+      String fProxyPort,
+      String fProxyBindTo,
+      String uriScheme,
+      String uriHost,
+      MultiValueTable<String, String> headers,
+      String resultUriPrefix) {
+    String schemeHostAndPort = UriFilterProxyHeaderParser.parse(
+        fakePortOption(fProxyPort),
+        fakeBindToOption(fProxyBindTo),
+        uriScheme,
+        uriHost,
+        headers)
+        .toString();
+    assertTrue(
+        String.format(
+            "schemeHostAndPort %s does not match expected %s; portConfig=\"%s\", bindTo=\"%s\", uriScheme=\"%s\", uriHost=\"%s\", headers=%s, expected=\"%s\"",
+            schemeHostAndPort,
+            resultUriPrefix,
+            fProxyPort,
+            fProxyBindTo,
+            uriScheme,
+            uriHost,
+            headers,
+            resultUriPrefix),
+        schemeHostAndPort.equals(resultUriPrefix));
+
+  }
+
+  private StringOption fakeBindToOption(String value) {
+    StringOption option = new StringOption(
+        new Config().createSubConfig("fake"),
+        "bindTo",
+        "127.0.0.1",
+        0,
+        false,
+        false,
+        "",
+        "",
+        new DummyStringCallback());
+    try {
+      option.setValue(value);
+    } catch (InvalidConfigValueException e) {
+      e.printStackTrace();
```

Comment:
No, no, no. This is auto-generated try-catch block, shame on you! :) Please let the exception bubble up and fail the test.

Reply:
*cowers in shame* … :-)

After doing that I remembered why I did not: Every test method now throws an Exception. But it’s more robust, so it did get better.